pkgs/applications/virtualization/gvisor/default.nix

```diff
@@ -0,0 +1,176 @@
+    rev   = "d97ccfa346d23d99dcbe634a10fa5d81b089100d";
+
+    # NOTE: this is the output of the whole fixed-output derivation, so
+    # `nix-prefetch-git` won't work to obtain this. The easiest way is to just
+    # change it and see what breaks :)
+    sha256 = "1bcnq7kazbf6l5j0g82x2lvg1nbp7z70klk139dxi0jkw0j8dh3r";
+
+    # Patch the source to:
+    #   - Use our host Go toolchain
+    #   - Fix hard-coded paths to Bash, coreutils, etc.
+    #   - Fetch a specific version of abseil instead of the `master` branch
+    extraPostFetch =
+      let
+        # master as of 2018-11-10
+        abseilRev  = "070f6e47b33a2909d039e620c873204f78809492";
+        abseilHash = "a4298dca4149157b379588bec19493bcab56f8b3fb119ea81303b04d70af1b48";
+
+      in ''
+        substituteInPlace "$out/WORKSPACE" \
+          --replace 'go_register_toolchains(go_version="1.11.2")' 'go_register_toolchains(go_version="host")' \
+          --replace \
+            'urls = ["https://github.com/abseil/abseil-cpp/archive/master.zip"],' \
+            'urls = ["https://github.com/abseil/abseil-cpp/archive/${abseilRev}.zip"], sha256 = "${abseilHash}",'
+
+        find "$out" -name '*.sh' -exec \
+          sed -i 's|#!/bin/bash|#!${bash}/bin/bash|g' {} \;
+
+        find "$out" -name '*_test.go' -exec \
+          sed -i \
+            -e 's|/bin/sleep|${coreutils}/bin/sleep|g' \
+            -e 's|/bin/true|${coreutils}/bin/true|g' \
+            {} \;
+      '';
+  };
+
+  # Bazel command we run.
+  bazelCmd = "USER=nix bazel";
+
+  # Use Bazel to fetch dependencies as a fixed-output derivation, so that we
+  # have network access.
+  bazelDependencies = stdenv.mkDerivation rec {
+    inherit version;
+    name = "gvisor-build-dependencies-${version}";
+
+    # The actual source
+    src = patchedSource;
+
+    outputHashMode = "recursive";
+    outputHashAlgo = "sha256";
+    outputHash = "0430pn3q71r6pyxq32k2n1zhnp9hvs5mizvw3zy6zwrsv3fchdb6";
+
+    nativeBuildInputs = [ bazel git go symlinks ];
+    GIT_SSL_CAINFO = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+
+    builder = writeScript "builder.sh" ''
+      source ${stdenv}/setup
+
+      # Copy sources into our build directory
+      cp -pr --reflink=auto -- "$src" ./source
+      chmod -R u+w -- ./source
+      cd source
+
+      # Prefetch dependencies with Bazel
+      export TEST_TMPDIR=$PWD/bazel_root_dir
+      ${bazelCmd} sync
+
+      outputUserRoot="$TEST_TMPDIR/_bazel_nix"
+      outputBase="$outputUserRoot/${outputBaseHash}"
+      if [[ ! -d "$outputBase" ]]; then
+        echo "did not find outputBase directory: $outputBase"
+        exit 1
+      fi
+
+      # Find git repositories and remove the `.git` directory entirely; Bazel
+      # appears to work just fine, and this is a major source of
+      # nondeterminism.
+      find "$outputBase/external" -name '.git' -type d -exec rm -rf {} '+'
+
+      # Convert symbolic links to relative ones
+      ( cd "$outputBase/external" && symlinks -cr . )
```

Comment:
Ooooh, I didn’t know about the `symlinks` utility/package. Smexy.

Reply:
Though the manpage of `symlinks` doesn’t mention a conversion from absolute to relative targets:

```
       -c

              a desirable setup in most cases.  This option also causes any messy links to  be  cleaned
              up, and, if -s was also specified, then lengthy links are also shortened.  Links affected
              by -c are prefixed with changed in the output.

       -r     recursively operate on subdirectories within the same filesystem.
```

Reply:
It's mentioned in the online manpage [here](https://linux.die.net/man/8/symlinks):

```
-c      convert absolute links (within the same filesystem) to relative links.
        This permits links to maintain their validity regardless of the mount
        point used for the filesystem -- a desirable setup in most cases. This
        option also causes any messy links to be cleaned up, and, if -s was
        also specified, then lengthy links are also shortened. Links affected
        by -c are prefixed with changed in the output.  
```

Reply:
You are right, the first two lines were missing in my output. Strange. When I open the version from master it seems fixed.